waspc/data/Generator/templates/react-app/src/router/routes.ts

```diff
@@ -0,0 +1,47 @@
+{{={= =}=}}
+export type Routes = 
+  {=# routes =}
+| {to: "{= urlPath =}", params{=^ urlParams =}?{=/ urlParams =}: {{=# urlParams =}{= . =}: string | number;{=/ urlParams =}}}
```

Comment:
Why are `params` always optional, even for routes that don't use them?

Reply:
Got it. Does this have implications on the functionality (i.e., will the user have worse type checking support because of it)?

Reply:
Now with the latest iteration, it's a bit different, but to answer the initial question => users get all the benefits they expect from type-safe routes.